tests/unit/test_control.py

```diff
@@ -24,43 +23,24 @@ def test_passing_host(self):
         (-1, [{"status": {"ready": False}}], 0, 0),
     ])
     def test_create_index_with_timeout(self, mocker, timeout_value, describe_index_responses, expected_describe_index_calls, expected_sleep_calls):
-        mocker.patch.object(IndexOperationsApi, 'describe_index', side_effect=describe_index_responses)
-        mocker.patch.object(IndexOperationsApi, 'create_index')
+        p = Pinecone(api_key="123-456-789")
+        mocker.patch.object(p.index_api, 'describe_index', side_effect=describe_index_responses)
+        mocker.patch.object(p.index_api, 'create_index')
         mocker.patch('time.sleep')
 
-        p = Pinecone(api_key="123-456-789")
         p.create_index("my-index", 10, timeout=timeout_value, cloud="aws", region="us-west1", capacity_mode="pod")
 
-        assert IndexOperationsApi.create_index.call_count == 1
-        assert IndexOperationsApi.describe_index.call_count == expected_describe_index_calls
+        assert p.index_api.create_index.call_count == 1
+        assert p.index_api.describe_index.call_count == expected_describe_index_calls
         assert time.sleep.call_count == expected_sleep_calls
 
     def test_create_index_when_timeout_exceeded(self, mocker):
         with pytest.raises(TimeoutError):
-            get_status_responses = [{"status": {"ready": False}}] * 5
-            mocker.patch.object(IndexOperationsApi, 'describe_index', side_effect=get_status_responses)
-            mocker.patch.object(IndexOperationsApi, 'create_index')
-            mocker.patch('time.sleep')
-
             p = Pinecone(api_key="123-456-789")
-            p.create_index("my-index", 10, timeout=10, cloud="aws", region="us-west1", capacity_mode="pod")
-
-    # @pytest.mark.parametrize("timeout_value, list_indexes_calls, time_sleep_calls, list_indexes_responses", [
```

Comment:
This commented code was duplicating a different test, so i just removed it.

Reply:
I thought it was duping `test_create_index_with_timeout` (similar parameterized setup, the word "timeout"), but actually now that I am looking closer I see it doing delete_index instead of create_index. I should add more tests here, but I can do that in a separate follow-up.